src/commands/CommandClear.java

```diff
@@ -22,31 +24,81 @@
 	@Override
 	public void action(){
 		
-		new CommandConfirmed(this){
+		if(!hasParameter("u","s","b")){
 			
-			@Override
-			public String getConfMessage(){
-				return lang("ConfirmMessage");
-			}
-			
-			@Override
-			public void confirmed(){
+			new CommandConfirmed(this){
 				
-				try{
-					
-					deleteAllMessages();
+				@Override
+				public String getConfMessage(){
+					return lang("ConfirmMessage");
+				}
+				
+				@Override
+				public void confirmed(){
 					
-					if(isAlive())
-						sendMessage("Cleared everything!");
+					try{
+						
+						deleteAllMessages();
+						
+						if(isAlive())
+							sendMessage("Cleared everything!");
+						
+					}
+					catch(PermissionException e){
+						sendMessage(lang("NoPermission"));
+					}
 					
 				}
-				catch(PermissionException e){
-					sendMessage(lang("NoPermission"));
-				}
 				
+			};
+		}
+		else{
+			
+			try{
+				final Member usr;
+				
+				if(hasParameter("b")){
```

Comment:
As per my previous comment, the order of the if/else block should be this : `-u`, `--self`, `--bot`

This way, if we get the request `!!clear -u @V-ed --bot`, we won't clear the bot's messages and `V-ed`'s instead. This would also make the request `!!clear --bot -u @V-ed` clear `V-ed`'s messages, but that's the priority I'm willing to put.

In this PR state, your order is this : `--bot`, `--self`, `-u`, which is not what I would expect.